src/graphql/account.rs

```diff
@@ -418,6 +418,51 @@ impl AccountMutation {
         update_jwt_expires_in(expires_in)?;
         Ok(time)
     }
+
+    /// Retrieves the user's language selection by username.
+    #[graphql(guard = "RoleGuard::new(super::Role::SystemAdministrator)
+        .or(RoleGuard::new(super::Role::SecurityAdministrator))
+        .or(RoleGuard::new(super::Role::SecurityManager))
+        .or(RoleGuard::new(super::Role::SecurityMonitor))")]
+    async fn language(&self, ctx: &Context<'_>, username: String) -> Result<Option<String>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.account_map();
+
+        map.get(&username)?
+            .ok_or_else::<async_graphql::Error, _>(|| "Account not found".into())
+            .map(|account| account.language)
+    }
+
+    /// Updates only the user's language setting, regardless of the user's role.
+    #[graphql(guard = "RoleGuard::new(super::Role::SystemAdministrator)
+        .or(RoleGuard::new(super::Role::SecurityAdministrator))
+        .or(RoleGuard::new(super::Role::SecurityManager))
+        .or(RoleGuard::new(super::Role::SecurityMonitor))")]
+    async fn update_language(
+        &self,
+        ctx: &Context<'_>,
+        username: String,
+        language: UpdateLanguage,
+    ) -> Result<String> {
+        if language.new.is_none() {
+            return Err("New language must be provided.".into());
+        }
+
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.account_map();
+
+        map.update(
+            username.as_bytes(),
+            &None,
+            None,
+            &None,
+            &None,
+            &Some((language.old, language.new)),
+            &None,
+            &None,
+        )?;
+        Ok(username)
```

Comment:
I think it would be better to return the new language rather than `username`.

Reply:
I have modified the code to return the new language value.